dev_guide/getting_traffic_into_cluster.adoc

```diff
@@ -0,0 +1,148 @@
+after that the users can self-service the ingress without having to bother
+the admins. The router has controls to allow the admin to specify whether
+the users can self-provision hostnames, or if they must fit a pattern the
+admin defines. The other solutions require the admin to do it, or it requires
+that they delegate a lot of privilege.
+
+A set of routes can be created in the various projects. The overall
+set of routes is available to the set of routers. Each router selects
+from the set of routes.  All routers see all routes unless restricted
+by labels on the router, this is called router
+xref:../architecture/core_concepts/routes.adoc:#router-sharding[sharding].
+
+The xref:../admin_guide/high_availability.adoc#[High Availability] section
+describes how to configure a router for high availbility service using
+multiple replicas.
+
+[[using-the-loadbalancer]]
+== Using a Service of Type LoadBalancer
+
+There are load balancers available on
+xref:../install_config/configuring_aws.adoc#[AWS] and
+xref:../install_config/configuring_gce.adoc#[GEC] clouds as well as
+xref:../admin_guide/tcp_ingress_external_ports.adoc#[bare metal].
+
+The bare-metal
+xref:../admin_guide/tcp_ingress_external_ports.adoc#[LoadBalancer] allocates
+a unique IP from a configured pool. This limits you to a single ingress IP
+(can be a VIP, but still will be a single machine for initial load balancing).
+It simplifies the admin job, but uses one IP per service.
+
+Load balancing is described
+xref:http://kubernetes.io/docs/user-guide/services/#type-loadbalancer[here].
+
+[[using-externalIP]]
+== Using a Service ExternalIP
+
+The user can manually assign
+xref:http://kubernetes.io/docs/user-guide/services/#external-ips[ExternalIPs]
+to the service. The supplied list of IP addresses are used for load balancing
+incoming requests. The service port is opened on all IP addresses in the list.
+Doing this requires elevated permissions to assign, and manual tracking of the
+IP:ports that are in use.
+
+An externally visible IP can be configured in one of several ways:
+
+- manually configure the externalIPs with a known external ip address.
+- configure externalIPs to a xref:../admin_guide/high_availability.adoc#[VIP]
+that is generated from (VRRP)
+- In a cloud (xref:../install_config/configuring_aws.adoc#[AWS] or
+xref:../install_config/configuring_gce.adoc#[GEC]) use type=LoadBalancer
+- In a xref:../admin_guide/tcp_ingress_external_ports.adoc#[bare metal]
+environment, configure an ingressIP range (IngressIPNetworkCIDR),
+service.type=LoadBalancer and service.port.ingressIP
+
+The admin must make sure the routes allow access to the nodes and local
+firewall rules on all nodes allow access to the open port.
+
+Use the same nodes as the router but advertise them with VRRP and use
+the DNS address of the router nodes.
+
+
+[[using-nodeport]]
+== Using a Service NodePort
+
+Use NodePorts to expose the service nodePort on all nodes in the cluster. This is
```

Comment:
Consider mentioning NodePorts limitation that ports are in the 30-60k range by default, which means a NodePort port is unlikely to match a service's intended port (e.g. 8080 might be exposed as 31020).